### ISSUE TITLE
Ensure dependent resources can only have a single parent

### ISSUE THREAD

**Issue by equill** [equill/restagraph#24]
This is a necessary prerequisite for being able to infer a definitive path to any given resource. It also strengthens the distinction between a thing that exists only in the context of another thing, and an independent thing that is most commonly found connected to a single other thing.

Apply this when re-homing a dependent resource, and when creating a relationship to an existing resource - refuse to create a dependent relationship outside of creating a dependent resource.

I'm 99% sure I wrote the code this way already, but I need to be certain.

- #77 depends on this

**Comment by equill** [comment]
This may work best by rejecting any attempts to define a relationship to a dependent resource with "many" at the parent's end, and making sure cardinality is solidly enforced.

**Comment by equill** [comment]
Added tests to check on this, for parents with the _same_ relationship.

Found and fixed a bug in the API dispatcher in the process.

**Comment by equill** [comment]
The second half of this challenge (parents with _different_ dependent relationships to the dependent type) overlaps with #77 - the most viable way of doing this is via a `canonical-path` function.

**Comment by equill** [comment]
This depends on https://github.com/equill/neo4cl/issues/15 for best results.

**Comment by equill** [comment]
In the interests of accomplishing this, I'm implementing the path-canonicalisation functionality for #77, just without adding it to the API.

...this may be misguided. I was thinking that the best way of doing this is to fetch the target resource's canonical path(s), but this is redundant if the following conditions hold:

- `create-relationship` returns an error if you try to create a dependent relationship that way.
    - This ensures that there are only two remaining ways of creating a dependent relationship.
- `create-dependent-resource` always creates a new resource; it does not seek out a new one.
- `move-dependent-resource` already ensures that the old dependent relationship has been removed; I added tests to confirm this for #87 

This work isn't wasted, though, because the `self` option for #87 requires that we find the canonical path for both the source and target resources. It's just not so useful here.
